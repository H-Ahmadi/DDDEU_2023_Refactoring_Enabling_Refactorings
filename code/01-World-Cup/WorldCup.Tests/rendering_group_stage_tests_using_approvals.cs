using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using FluentAssertions;
using WorldCup.Model;
using WorldCup.Tests.TestDoubles;
using WorldCup.View;

namespace WorldCup.Tests;

//Approach 03 - Approval Testing
public class rendering_group_stage_tests_using_approvals
{

    [UseReporter(typeof(DiffReporter))]
    [Fact]
    public void Showing_a_single_group_stage_on_console()
    {
        var fakeConsole = new FakeConsole();

        var france = new Team("France");
        var australia = new Team("Australia");
        var denmark = new Team("Denmark");
        var tunisia = new Team("Tunisia");

        var group = new GroupStage()
        {
            Name = "FIFA WorldCup 2022 - Group D",
            Teams = new List<Team>() { france, australia, denmark, tunisia },
            Games = new List<Game>()
            {
                new Game(denmark, tunisia) { Score1 = 0, Score2 = 0 },
                new Game(france, australia) { Score1 = 4, Score2 = 1 },
                new Game(tunisia, australia) { Score1 = 0, Score2 = 1 },
                new Game(france, denmark) { Score1 = 2, Score2 = 1 },
                new Game(australia, denmark) { Score1 = 1, Score2 = 0 },
                new Game(tunisia, france) { Score1 = 1, Score2 = 0 },
            }
        };
        var service = new GroupViewService(fakeConsole);
        service.ShowGroup(group);
        
        var output = fakeConsole.GetOutput();
        Approvals.Verify(output);

    }
}