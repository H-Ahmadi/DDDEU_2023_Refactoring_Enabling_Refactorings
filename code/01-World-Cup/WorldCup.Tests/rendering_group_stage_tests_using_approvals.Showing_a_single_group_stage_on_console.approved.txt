+------------------------------------+------+------+------+------+------+------+-------------+
|                Team                |   P  |   W  |   D  |   L  |  GD  |  GS  |   Points    |
+------------------------------------+------+------+------+------+------+------+-------------+
|               France               |   3  |   2  |   0  |   1  |   3  |   6  |      6      |
|              Australia             |   3  |   2  |   0  |   1  |  -1  |   3  |      6      |
|               Tunisia              |   3  |   1  |   1  |   1  |   0  |   1  |      4      |
|               Denmark              |   3  |   0  |   1  |   2  |  -2  |   1  |      1      |
+------------------------------------+------+------+------+------+------+------+-------------+
